app/models/course/discussion/post.rb

```diff
@@ -98,4 +99,12 @@ def parent_topic_consistency
   def reparent_children
     children.update_all(parent_id: parent_id)
   end
+
+  # Should be called only when destroyed by association.
+  #
+  # We unset the children's parent id so they don't trigger a foreign key exception when the
+  # parent is marked for destruction first. They will be destroyed by association later.
```

Comment:
Possible to add 1 more assumption here, that when `destroyed_by_association` is `true`, we assume that the entire `topic` will be deleted?